Send empty arrays properly

Guardiola31337 opened the pull request:
- Fixes #337 

👀 @ericrwolfe @cammace

Guardiola31337 commented:
Addressed in #354 👇
Assuming:
- `beforeLocations` or `afterLocations`
  - If `null` 👉 `null`
  - If empty 👉 `[]`
  - If not empty 👉 `[{}]`

Is that correct ☝️ @ericrwolfe? If not, let me know. Should be easy to tweak.

cc @cammace

ericrwolfe commented:
> - `beforeLocations` or `afterLocations`
>   - If `null` 👉 `null`
>   - If empty 👉 `[]`
>   - If not empty 👉 `[{}]`

@Guardiola31337 Only change needed here would be:
- If `null` 👉 `[]`

Guardiola31337 commented:
https://github.com/mapbox/mapbox-navigation-android/pull/354/commits/bdc08514e9783adb16db9279c64424caff3bce62 addressed your comment @ericrwolfe 
Ready for a second round of 👀
